trucks.js

```diff
@@ -501,4 +501,196 @@ var foodTrucks = [
 		Facebook: 'https://www.facebook.com/fulltilticecreams',
 		Twitter: 'https://twitter.com/fticecream'
 	}
-];
\ No newline at end of file
+];
+
+/*
+ * Homework 
+ * JSCRIPT300-Spring2015/Module_4b
+ * by Diane Zevenbergen
+ *
+ * NOTE: I revised code from HW for Module 4a in order to use
+ * underscore in getTruck, getFoodTypes and filterByFoodType
+ */
+
+'use strict';
+
+var _ = require('underscore');
+
+/* 
+ * Function returns array object of food trucks available today.
+ * Uses underscore's filter() method.
+ */ 
+function filterByDay(day) {
+
+    // Filter returns the subset of foodTrucks that match 'day'
+    var todaysTrucks = _.filter(foodTrucks, function(item) {
+
+        // Exclude truck if object doesn't include 'schedule' prop, otherwise check for 'day' match
+        if (typeof item.schedule === 'undefined') {
+            return false;
+        } else {
+            return (item.schedule.indexOf(day) !== -1);
+        }        
+    });
+    
+    return todaysTrucks;
+}
+
+// Function returns all truck objects
+function getTrucks() {
+    return foodTrucks;
+}
+
+// Function returns object info for matching truck if found
+function getTruck(name) {
+    var matchingTruck = {};
+    
+    name = name.toLowerCase().trim();
+
+    // Use underscore to find matching truck
+    matchingTruck = _.find(foodTrucks, function (i) { 
+        if (i.name.toLowerCase().trim() === name) {
```

Comment:
Just return the result of the conditional. Like many other _ functions, the function is just used to return true or false and the function will return the object based on the result of that truth test.